bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageHandler.java

```diff
@@ -0,0 +1,155 @@
+import java.io.IOException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The {@link OpenGarageHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Smedley - Initial contribution
+ */
+
+@NonNullByDefault
+public class OpenGarageHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenGarageHandler.class);
+
+    private long refreshInterval;
+
+    private @Nullable OpenGarageWebTargets webTargets;
+    private @Nullable ScheduledFuture<?> pollFuture;
+
+    public OpenGarageHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            handleCommandInternal(channelUID, command);
+        } catch (OpenGarageCommunicationException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+        }
+    }
+
+    private void handleCommandInternal(ChannelUID channelUID, Command command) throws OpenGarageCommunicationException {
+        logger.warn("Received command {} for thing '{}' on channel {}", command, thing.getUID().getAsString(),
+                    channelUID.getId());
+        switch (channelUID.getId()) {
+            case OpenGarageBindingConstants.CHANNEL_OG_STATUS:
+                if (command instanceof OnOffType) {
+                    changeStatus(((OnOffType) command).equals(OnOffType.ON));
+                    return;
+                }
+                break;
+        }
+
+        logger.warn("Received command {} of wrong type for thing '{}' on channel {}", command, thing.getUID().getAsString(),
+                    channelUID.getId());
+    }
+
+    @Override
+    public void initialize() {
+        OpenGarageConfiguration config = getConfigAs(OpenGarageConfiguration.class);
+        logger.debug("config.hostname = {}, refresh = {}, port = {}", config.hostname, config.refresh, config.port);
+        if (config.hostname == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Hostname/IP address must be set");
+        } else {
+            webTargets = new OpenGarageWebTargets(config.hostname, config.port, config.password);
+            refreshInterval = config.refresh;
+
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        logger.debug("Scheduling poll for 500ms out, then every {} s", refreshInterval);
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 1, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(true);
```

Comment:
Doesn’t this produce a null warning?

Reply:
Function is removed now (per above). Fixed in 1641769b62